# Exploratory Data Analysis (EDA) report

## 1. Report overview

### 1.1. Project Title: 	**Lisbon Airbnb analysis, december 2023**

![Lisboa menina e moça](https://images.resosys.com/destinations/10/articles/getting-around_en/inline/1666700246-svetlana-gumerova-golcaouc7ia-unsplash.jpg)

### 1.2. Context:

Airbnb is an online marketplace for people to rent places to stay all over the world.

Lisbon is the capital city of Portugal, located in Europe. Situated along the Atlantic coast, Lisbon offers a rich maritime heritage and a mild Mediterranean climate, making it an attractive destination for travelers worldwide, and, consequently, for real state investors.

As a real state investor, I am interested in acquiring a property in Lisbon. I want to know in which Lisbon area I should invest and which requisites are important to take into consideration when charging higher prices to tourists.


### 1.3. Goal:

In this EDA we are going to focus on 2 questions:

1 - Where should I invest based on price and property quantity?

2 - Does review rating influence property prices?

### 1.4. Dataset used:

http://insideairbnb.com/get-the-data/     Lisbon, Portugal, December 2023

listings.csv.gz - 	Detailed Listings data


## 2. Data Overview  - TESTETO SEE IT IS COMMITING

### 2.1. Import libraries


In [ ]:
import pandas as pd
from matplotlib import pyplot as plt
import seaborn as sns
import folium
import numpy as np

### 2.2 Load data

In [182]:
listings = pd.read_csv('./data/listings.csv.gz')
listings

id                                       listing_url  \
0                     6499                 https://www.airbnb.com/rooms/6499   
1                   199461               https://www.airbnb.com/rooms/199461   
2                    25659                https://www.airbnb.com/rooms/25659   
3                    29248                https://www.airbnb.com/rooms/29248   
4                   208683               https://www.airbnb.com/rooms/208683   
...                    ...                                               ...   
22746  1045316044871166129  https://www.airbnb.com/rooms/1045316044871166129   
22747  1045433384003538979  https://www.airbnb.com/rooms/1045433384003538979   
22748  1045443962999171278  https://www.airbnb.com/rooms/1045443962999171278   
22749  1045468693077800153  https://www.airbnb.com/rooms/1045468693077800153   
22750  1045570587337952964  https://www.airbnb.com/rooms/1045570587337952964   

            scrape_id last_scraped       source  \
0      20231217045130   2023-12-17  city scrape   
1      20231217045130   2023-12-18  city scrape   
2      20231217045130   2023-12-18  city scrape   
3      20231217045130   2023-12-17  city scrape   
4      20231217045130   2023-12-18  city scrape   
...               ...          ...          ...   
22746  20231217045130   2023-12-18  city scrape   
22747  20231217045130   2023-12-18  city scrape   
22748  20231217045130   2023-12-17  city scrape   
22749  20231217045130   2023-12-17  city scrape   
22750  20231217045130   2023-12-18  city scrape   

                                                    name  description  \
0      Rental unit in Lisbon · ★4.49 · 1 bedroom · 3 ...          NaN   
1      Rental unit in Lisboa · ★4.92 · 2 bedrooms · 3...          NaN   
2      Rental unit in Lisboa · ★4.82 · 1 bedroom · 1 ...          NaN   
3      Rental unit in Lisbon · ★4.57 · 1 bedroom · 2 ...          NaN   
4      Rental unit in Lisbon · ★4.68 · 2 bedrooms · 2...          NaN   
...                                                  ...          ...   
22746  Rental unit in Lisboa · ★New · 2 bedrooms · 1 ...          NaN   
22747  Rental unit in Lisboa · ★New · 2 bedrooms · 3 ...          NaN   
22748  Rental unit in Lisboa · ★New · 1 bedroom · 1 b...          NaN   
22749  Tent in Lisboa · ★New · 2 bedrooms · 3 beds · ...          NaN   
22750  Rental unit in Lisboa · ★New · 1 bedroom · 2 b...          NaN   

                                   neighborhood_overview  \
0      To get  to the city center, tram 15 stops in f...   
1      The BAIXA (down town, lower area) neighbourhoo...   
2      The Alfama neighborhood, is the oldest and mos...   
3                                                    NaN   
4      O bairro de Alfama é um dos bairros mais antig...   
...                                                  ...   
22746                                                NaN   
22747  Apartment is located in one of the most touris...   
22748  Marquês de Pombal Square, often simply referre...   
22749                                                NaN   
22750                                                NaN   

                                             picture_url    host_id  \
0      https://a0.muscache.com/pictures/6422ee92-c84e...      14455   
1      https://a0.muscache.com/pictures/prohost-api/H...     914126   
2      https://a0.muscache.com/pictures/miso/Hosting-...     107347   
3      https://a0.muscache.com/pictures/4212cc65-2046...     125768   
4      https://a0.muscache.com/pictures/21533977/a19d...     753902   
...                                                  ...        ...   
22746  https://a0.muscache.com/pictures/miso/Hosting-...  207402423   
22747  https://a0.muscache.com/pictures/miso/Hosting-...   62521369   
22748  https://a0.muscache.com/pictures/prohost-api/H...    1756107   
22749  https://a0.muscache.com/pictures/miso/Hosting-...  494418680   
22750  https://a0.muscache.com/pictures/miso/Hosting-...   62521369   


### 2.3. Get to know the data

In [65]:
listings.head()

id                          listing_url       scrape_id last_scraped  \
0    6499    https://www.airbnb.com/rooms/6499  20231217045130   2023-12-17   
1  199461  https://www.airbnb.com/rooms/199461  20231217045130   2023-12-18   
2   25659   https://www.airbnb.com/rooms/25659  20231217045130   2023-12-18   
3   29248   https://www.airbnb.com/rooms/29248  20231217045130   2023-12-17   
4  208683  https://www.airbnb.com/rooms/208683  20231217045130   2023-12-18   

        source                                               name  \
0  city scrape  Rental unit in Lisbon · ★4.49 · 1 bedroom · 3 ...   
1  city scrape  Rental unit in Lisboa · ★4.92 · 2 bedrooms · 3...   
2  city scrape  Rental unit in Lisboa · ★4.82 · 1 bedroom · 1 ...   
3  city scrape  Rental unit in Lisbon · ★4.57 · 1 bedroom · 2 ...   
4  city scrape  Rental unit in Lisbon · ★4.68 · 2 bedrooms · 2...   

   description                              neighborhood_overview  \
0          NaN  To get  to the city center, tram 15 stops in f...   
1          NaN  The BAIXA (down town, lower area) neighbourhoo...   
2          NaN  The Alfama neighborhood, is the oldest and mos...   
3          NaN                                                NaN   
4          NaN  O bairro de Alfama é um dos bairros mais antig...   

                                         picture_url  host_id  ...  \
0  https://a0.muscache.com/pictures/6422ee92-c84e...    14455  ...   
1  https://a0.muscache.com/pictures/prohost-api/H...   914126  ...   
2  https://a0.muscache.com/pictures/miso/Hosting-...   107347  ...   
3  https://a0.muscache.com/pictures/4212cc65-2046...   125768  ...   
4  https://a0.muscache.com/pictures/21533977/a19d...   753902  ...   

  review_scores_communication review_scores_location review_scores_value  \
0                        4.94                   4.80                4.39   
1                        4.93                   4.97                4.87   
2                        4.95                   4.88                4.82   
3                        4.75                   4.83                4.61   
4                        4.87                   4.90                4.66   

     license instant_bookable calculated_host_listings_count  \
0        NaN                t                              1   
1  114187/AL                f                             16   
2  56539/AL.                t                              1   
3   23535/AL                t                              5   
4   35037/AL                t                              6   

  calculated_host_listings_count_entire_homes  \
0                                           1   
1                                          16   
2                                           1   
3                                           5   
4                                           6   

  calculated_host_listings_count_private_rooms  \
0                                            0   
1                                            0   
2                                            0   
3                                            0   
4                                            0   

  calculated_host_listings_count_shared_rooms reviews_per_month  
0                                           0              0.56  
1                                           0              1.34  
2                                           0              1.69  
3                                           0              2.36  
4                                           0              0.98  

[5 rows x 75 columns]

In [66]:
#size
listings.shape

(22751, 75)

In [67]:
#Columns:
listings.columns

Index(['id', 'listing_url', 'scrape_id', 'last_scraped', 'source', 'name',
       'description', 'neighborhood_overview', 'picture_url', 'host_id',
       'host_url', 'host_name', 'host_since', 'host_location', 'host_about',
       'host_response_time', 'host_response_rate', 'host_acceptance_rate',
       'host_is_superhost', 'host_thumbnail_url', 'host_picture_url',
       'host_neighbourhood', 'host_listings_count',
       'host_total_listings_count', 'host_verifications',
       'host_has_profile_pic', 'host_identity_verified', 'neighbourhood',
       'neighbourhood_cleansed', 'neighbourhood_group_cleansed', 'latitude',
       'longitude', 'property_type', 'room_type', 'accommodates', 'bathrooms',
       'bathrooms_text', 'bedrooms', 'beds', 'amenities', 'price',
       'minimum_nights', 'maximum_nights', 'minimum_minimum_nights',
       'maximum_minimum_nights', 'minimum_maximum_nights',
       'maximum_maximum_nights', 'minimum_nights_avg_ntm',
       'maximum_nights_avg_ntm', 'ca

In [198]:
#Check type of columns and other info: - I used this method to see all columns. With dtypes, it only shows the 5 first and 5 last columns.
listings.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 22751 entries, 0 to 22750
Data columns (total 75 columns):
 #   Column                                        Non-Null Count  Dtype  
---  ------                                        --------------  -----  
 0   id                                            22751 non-null  int64  
 1   listing_url                                   22751 non-null  object 
 2   scrape_id                                     22751 non-null  int64  
 3   last_scraped                                  22751 non-null  object 
 4   source                                        22751 non-null  object 
 5   name                                          22751 non-null  object 
 6   description                                   0 non-null      float64
 7   neighborhood_overview                         13390 non-null  object 
 8   picture_url                                   22751 non-null  object 
 9   host_id                                       22751 non-null 

In [125]:
# general info:
listings.describe()

id     scrape_id  description       host_id  \
count  2.275100e+04  2.275100e+04          0.0  2.275100e+04   
mean   3.214074e+17  2.023122e+13          NaN  1.764003e+08   
std    4.152084e+17  0.000000e+00          NaN  1.765189e+08   
min    6.499000e+03  2.023122e+13          NaN  1.405900e+04   
25%    2.092871e+07  2.023122e+13          NaN  2.120087e+07   
50%    4.285647e+07  2.023122e+13          NaN  1.094943e+08   
75%    7.837875e+17  2.023122e+13          NaN  3.032899e+08   
max    1.047514e+18  2.023122e+13          NaN  5.509970e+08   

       host_listings_count  host_total_listings_count      latitude  \
count         22751.000000               22751.000000  22751.000000   
mean             32.696189                  45.252297     38.762083   
std             112.744673                 233.508810      0.112169   
min               1.000000                   1.000000     38.676140   
25%               2.000000                   2.000000     38.711030   
50%               5.000000                   6.000000     38.719080   
75%              17.000000                  21.000000     38.747326   
max            2641.000000                8868.000000     39.303860   

          longitude  accommodates  bathrooms   bedrooms          beds  \
count  22751.000000  22751.000000        0.0   6.000000  22538.000000   
mean      -9.208175      3.866423        NaN   6.500000      2.472003   
std        0.112812      2.376913        NaN   4.135215      2.025099   
min       -9.488100      1.000000        NaN   2.000000      1.000000   
25%       -9.291569      2.000000        NaN   4.250000      1.000000   
50%       -9.147670      4.000000        NaN   6.000000      2.000000   
75%       -9.134860      5.000000        NaN   7.000000      3.000000   
max       -8.838504     16.000000        NaN  14.000000     76.000000   

       minimum_nights  maximum_nights  minimum_minimum_nights  \
count    22751.000000    22751.000000            22751.000000   
mean         4.453299      554.078678                4.617379   
std         18.013646      522.322018               20.883668   
min          1.000000        1.000000                1.000000   
25%          1.000000       90.000000                1.000000   
50%          2.000000      365.000000                2.000000   
75%          3.000000     1125.000000                3.000000   
max       1000.000000    36180.000000             1000.000000   

       maximum_minimum_nights  minimum_maximum_nights  maximum_maximum_nights  \
count            22751.000000            22751.000000            22751.000000   
mean                 9.417872              661.046591              735.937541   
std                 40.464018              538.354827              519.240736   
min                  1.000000                1.000000                1.000000   
25%                  2.000000              150.000000              365.000000   
50%                  3.000000              999.000000             1125.000000   
75%                  4.000000             1125.000000             1125.000000   
max               1000.000000            36180.000000            36180.000000   

       minimum_nights_avg_ntm  maximum_nights_avg_ntm  calendar_updated  \
count            22751.000000            22751.000000               0.0   
mean                 7.713112              691.716742               NaN   
std                 36.551113              522.258488               NaN   
min                  1.000000                1.000000               NaN   
25%                  2.000000              317.700000               NaN   
50%                  2.400000              999.000000               NaN   
75%                  3.600000             1125.000000               NaN   
max               1000.000000            36180.000000               NaN   

       availability_30  availability_60  availability_90  availability_365  \
count     22751.000000     22751.000000     22751.000000      22751.0000

In [83]:
#Count values per neighbourhood_group:
listings.value_counts("neighbourhood_group_cleansed")

neighbourhood_group_cleansed
Lisboa                   15611
Cascais                   2149
Sintra                    1537
Mafra                     1289
Lourinh                    481
Oeiras                     437
Torres Vedras              327
Loures                     281
Amadora                    234
Odivelas                   133
Alenquer                    87
Vila Franca De Xira         74
Cadaval                     59
Sobral De Monte Agrao       28
Azambuja                    14
Arruda Dos Vinhos           10
Name: count, dtype: int64

In [164]:
#Lisbon Map
#define coordinates based on min and max values of data: latitude = 38.676140, longitude = -9.488100	
lisbon_coords = [38.9, -9.139480277813723]

map = folium.Map(
    location = lisbon_coords,
    zoom_start=11, 
    tiles="OpenStreetMap",
)
map

In [177]:
#Lisbon Heat Map - quantity of properties
from folium.plugins import HeatMap

map.add_child(
    HeatMap(
        listings[['latitude','longitude']],
    )
)

In [82]:
#Count null values
# is null().sum() or null().any()

nan_columns = listings.columns[listings.isnull().any()]

nan_sum = listings[nan_columns].isnull().sum()

print(nan_sum)


description                    22751
neighborhood_overview           9361
host_location                   5957
host_about                      9862
host_response_time              3103
host_response_rate              3103
host_acceptance_rate            1813
host_is_superhost                182
host_neighbourhood             13387
neighbourhood                   9361
bathrooms                      22751
bathrooms_text                    28
bedrooms                       22745
beds                             213
price                           1161
calendar_updated               22751
has_availability                1161
first_review                    2953
last_review                     2953
review_scores_rating            2943
review_scores_accuracy          2945
review_scores_cleanliness       2945
review_scores_checkin           2945
review_scores_communication     2945
review_scores_location          2947
review_scores_value             2944
license                         1397
r

In [166]:
#Count duplicates

#get duplicated rows:
listings[listings.duplicated()]


Empty DataFrame
Columns: [id, listing_url, scrape_id, last_scraped, source, name, description, neighborhood_overview, picture_url, host_id, host_url, host_name, host_since, host_location, host_about, host_response_time, host_response_rate, host_acceptance_rate, host_is_superhost, host_thumbnail_url, host_picture_url, host_neighbourhood, host_listings_count, host_total_listings_count, host_verifications, host_has_profile_pic, host_identity_verified, neighbourhood, neighbourhood_cleansed, neighbourhood_group_cleansed, latitude, longitude, property_type, room_type, accommodates, bathrooms, bathrooms_text, bedrooms, beds, amenities, price, minimum_nights, maximum_nights, minimum_minimum_nights, maximum_minimum_nights, minimum_maximum_nights, maximum_maximum_nights, minimum_nights_avg_ntm, maximum_nights_avg_ntm, calendar_updated, has_availability, availability_30, availability_60, availability_90, availability_365, calendar_last_scraped, number_of_reviews, number_of_reviews_ltm, number_of_reviews_l30d, first_review, last_review, review_scores_rating, review_scores_accuracy, review_scores_cleanliness, review_scores_checkin, review_scores_communication, review_scores_location, review_scores_value, license, instant_bookable, calculated_host_listings_count, calculated_host_listings_count_entire_homes, calculated_host_listings_count_private_rooms, calculated_host_listings_count_shared_rooms, reviews_per_month]
Index: []

In [167]:
#count duplicated rows:
listings.duplicated().sum()

0

### 2.3.1. Conclusion of this section:

- We have 75 columns and 22751 rows.

- With the heat map we see the in Lisbon district that the data is more concentraded around Lisbon center, in the south of the north bay and along the coast. For this study, only the city center will be included. The ramaining rows won't taken into consideration.

- There are no duplicated rows.

- There are several columns with null values, including price. There is 1161 rows which price is null. This rows should be deleted from the dataset.

- There are columns with all values null. This rows should be deleted as well.

- Price is an object type. It should be set to float before ploting and analysing the data based on price.



## 3. Data Cleaning and Preparation

### 3.1. Delete NaN price
- **Outlier Detection & Treatment**:
- **Feature Engineering**: